### PR TITLE
fix: Import/upload unstable tests

### PR DESCRIPTION
**Pull request by michaljurecko**
Slack: https://keboolaglobal.slack.com/archives/C055WKPHYSD/p1725887204675479

**Changes:**
- Fixed unstable tests: import/upload/E2E.
- Fixed minor bug about `waitForSlicesUpload` error.

-----------

### CODE REVIEW

**Comment by Matovidlo** [comment]
@michaljurecko este E2E test asi sa zmenil skrz zmenu logovania

**Comment by michaljurecko** [comment]
@Matovidlo I know, it seems rather that there is some instability there, it works locally. I'm looking into it.

File: internal/pkg/service/stream/storage/node/coordinator/filerotation/operator.go [hunk 1]
```diff
@@ -372,67 +373,68 @@ func (o *operator) closeFile(ctx context.Context, file *fileData) {
...
-	o.logger.Info(ctx, "closing file")
+	// Wait for all slices upload, get statistics
+	stats, opErr := o.waitForFileClosing(ctx, file)
 
-	// Wait until all file slices are uploaded
-	if err := o.waitForSlicesUpload(ctx, file.FileKey); err != nil {
-		err = errors.PrefixError(err, "error when waiting for file slices upload")
-		o.logger.Error(ctx, err.Error())
-		return
-	}
```
Comment (michaljurecko) [inline internal/pkg/service/stream/storage/node/coordinator/filerotation/operator.go:382]: There was some stuck slice upload (another/fixed bug), and the file import was therefore blocked.

But error from `waitForSlicesUpload` was not increasing retry attempt, so it was spamming logs every second.

File: test/stream/bridge/keboola/keboola_test.go [hunk 2]
```diff
@@ -82,162 +65,249 @@ func TestKeboolaBridgeWorkflow(t *testing.T) {
...
+	// Check initial state
+	ts.checkState(t, ctx, []file{
+		{
+			state: model.FileWriting,
+			volumes: []volume{
+				{
+					slices: []model.SliceState{
+						model.SliceWriting,
+					},
+				},
```
Comment (michaljurecko) [inline test/stream/bridge/keboola/keboola_test.go:77]: E2E test streams/Keboola was unstable and there were several issues in it - like we didn't finish properly under time stress, ... there was a lot of complicated math about which slice index is checked and so on.

I suggest to ook at the entire test, not the diff.

It should be much clearer what is happening and how the conditions are changing.

**Review by Matovidlo — APPROVED** [review]

File: internal/pkg/service/stream/storage/node/coordinator/filerotation/operator.go [hunk 3]
```diff
@@ -372,67 +373,68 @@ func (o *operator) closeFile(ctx context.Context, file *fileData) {
...
-		rErr := o.storage.File().IncrementRetryAttempt(file.FileKey, o.clock.Now(), err.Error()).RequireLock(lock).Do(ctx).Err()
+	if opErr != nil {
+		o.logger.Error(dbCtx, opErr.Error())
+		rErr := o.storage.File().IncrementRetryAttempt(file.FileKey, o.clock.Now(), opErr.Error()).RequireLock(lock).Do(ctx).Err()
 		if rErr != nil {
-			o.logger.Errorf(ctx, "cannot increment file close retry", err)
-			return
+			o.logger.Errorf(ctx, "cannot increment file close retry", rErr)
 		}
+		return
```
[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/service/stream/storage/node/coordinator/filerotation/operator.go:404]: It should be set, I thought we solved that already, I didn't check it, thanks for the check.

[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/service/stream/storage/node/coordinator/filerotation/operator.go:404]: Sure, I see the change, I'll think about where I copied it from :thinking:

File: internal/pkg/service/stream/storage/node/coordinator/filerotation/operator.go [hunk 4]
```diff
@@ -372,67 +373,68 @@ func (o *operator) closeFile(ctx context.Context, file *fileData) {
...
-		o.logger.Error(ctx, err.Error())
-		return
-	}
-
-	// Get file statistics
-	stats, err := o.statisticsCache.FileStats(ctx, file.FileKey)
-	if err != nil {
-		o.logger.Errorf(ctx, "cannot get file statistics: %s", err)
-		return
-	}
```
[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/service/stream/storage/node/coordinator/filerotation/operator.go:396]: Here it is important that the statistics can be loaded only after the waiting for slices.

Yes, we can add backoff delay also for other errors.

[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/service/stream/storage/node/coordinator/filerotation/operator.go:396]: Fixed/unified.

File: internal/pkg/service/stream/storage/node/coordinator/filerotation/operator.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/service/stream/storage/node/coordinator/filerotation/operator.go]: Fixed/unified. Thanks!

**Pull request merged**